### ISSUE TITLE
Every guard built on `endsWithInlineComment` declines a fix under Sass where a form feed closes the comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#284]
`endsWithInlineComment` reads a line feed and a carriage return as the breaks that close an inline comment, and leaves a form feed out. The comment beside that reading says why, and the reason is sound as far as it goes: Sass ends such a comment on a form feed and Less does not, and taking a comment for one still open only ever holds a fix back rather than letting a breaking one through.

It is still one answer where the two syntaxes give two, which is the fork [#132](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/132) had to split for the parentheses guard: its repair was to ask under readings a syntax actually has, rather than under a pole neither syntax stands at. Here the syntax is known — `readsInlineComments` probes it already — and what every caller gets is the intersection of the two readings.

Under `postcss-scss` the form feed does close the comment, and the two syntaxes can be measured against each other in two lines:

```js
scss.parse(`a { b: 1px // c\f\t2px; }`).first.first.value  // `1px \f\t2px` — the comment is gone
less.parse(`a { b: 1px // c\f\t2px; }`).first.first.value  // `1px // c\f\t2px` — the comment runs on
```

So wherever a file spells its breaks with form feeds, a fix that is safe under Sass is declined, and the warning survives every run of `--fix` without a byte being written.

## What it costs today

The `twins` oracle spells the breaks of each fixture three other ways and asks whether the rule says the same of the twin as of the original. Thirty of its rows are this defect, over six pairs of a rule and an option:

| Rule | Option | Rows under `postcss-scss` | Rows under `postcss-less` |
| --- | --- | --- | --- |
| `block-closing-brace-space-before` | `always` | 1 | 1 |
| `block-closing-brace-space-before` | `always-multi-line` | 1 | 1 |
| `block-closing-brace-space-before` | `never` | 4 | 3 |
| `block-closing-brace-space-before` | `never-multi-line` | 4 | 3 |
| `block-closing-brace-newline-before` | `never-multi-line` | 4 | 3 |
| `declaration-block-semicolon-space-before` | `always` | 3 | 2 |

The seventeen rows under `postcss-scss` are declines that are wrong. The thirteen under `postcss-less` are declines that are right, and the row is only the oracle noting that two differently spelled files genuinely differ.

The last pair came in with [#211](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/211), which gave this family the shared name `writesIntoInlineComment`; the other five with [#231](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/231), which gave that name three more callers and taught it to read the whitespace a node ends with as well. Every further caller multiplies the count again, and [#232](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/232), [#248](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/248), [#280](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/280) and [#281](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/281) are the four waiting to.

The remaining form-feed rows of that oracle are not this: `indentation`, `max-line-length`, `no-missing-end-of-source-newline` and `no-multiple-whitespaces` read a break rather than a comment, and the five rows of the `function-comma-*` rules all stand under `postcss-less`, where the reading they get is the right one.

## What would end it

`readsInlineComments` keeps two verdicts against each syntax, `spells` and `keeps`, and reads both off a single probe. Whether the syntax ends such a comment on a form feed is one more question of the same probe, and `endsWithInlineComment` would then read the breaks the syntax it was handed actually reads. Every caller already passes that answer along, so no signature has to grow a parameter: what is passed becomes the verdict rather than one boolean out of it.

Fixing this is fixing it everywhere at once, since all eleven callers ask through the same two utilities.

**Comment by firefoxic** [comment]
Merged into #333, which is the census of this and of two more levels of the same chain: #295, where the wrapper `writesIntoInlineComment` makes one asking of this scan, and #311, where the two `function-parentheses-*-inside` rules make two askings and put both to a file whose language is already known.

Nothing measured here is withdrawn. The thirty rows of the `twins` oracle, seventeen of them wrong declines under `postcss-scss`, are carried over and re-measured on `873cd12`, and so is the reading of `readsInlineComments` that ends it: the probe already keeps two verdicts off one parse, and whether the syntax closes a comment on a form feed is a third question of the same probe.

What the three had apart was three descriptions of one fork. Taken as one, the scan reads the breaks its verdict names, the wrapper passes the verdict along, and the two rules stop asking twice — one change rather than three that each explain why they are not the other two.
